src/Connection.php

```diff
@@ -44,6 +60,9 @@ public function __construct($properties = [])
                 ->addMoreInfo('properties', $properties);
         }
 
+        // backward compatibility - to be removed with Connection::$driverType
+        $this->driverType = array_search(static::class, self::$connectionRegistry, true);
```

Comment:
why is `$driverType` set also in Connection classes, can it be removed there?

Reply:
You are right, I was working on using classes only without `driverType` and this is left-over

Reply:
I actually liked this approach - do we need `defaultDriverType()` as all?

anyway, we can still implement `defaultDriverType()` like here and actually call that method here ;-)

and then we can remove all `public $driverType = 'xxx';`

The goal is to never specify driverType twice (once in registry, once in connection class).

Reply:
At the moment the issue comes with nested connections (like ProxyConnection) where driverType comes from the nested connection.
The idea of having driverType is not changed from my code so better to plan such changes in another PR

Reply:
Yes - I propose here to keep this removed line and remove all manual assigment. Once a connection class is registered, the effect will be the same.

Reply:
@georgehristov I went thru other feedback and this seems the last one to be addressed